frontend-react/src/network/api/NewApi.ts

```diff
@@ -0,0 +1,110 @@
+import { AxiosRequestConfig, AxiosRequestHeaders, Method } from "axios";
+
+/* Available APIs from the RS API */
+export enum ApiBaseUrls {
+    HISTORY = "/api/history", // SAMPLE
+    LOOKUP_TABLES = "/api/lookuptables",
+}
+export type AvailableMethods = Method[];
+export interface Endpoint {
+    url: string;
+    methods: AvailableMethods;
+}
+/* Name your endpoints:
+ * e.g. new Map([
+ *   ["allOrgs", { url: "/organization", methods [GET] }],
+ *   ["org", { url: "/organization/:id", methods: [GET, UPDATE] }]
+ * ]) */
+export type EndpointMap = Map<string, Endpoint>;
+/* Declaration of an API */
+export interface API {
+    baseUrl: ApiBaseUrls;
+    endpoints: EndpointMap;
+}
+/* Make some headers required */
+export interface RSRequestHeaders extends AxiosRequestHeaders {}
+/* Make some fields required or overwrite types */
+export interface RSRequestConfig extends AxiosRequestConfig {
+    url: string;
+    method: Method;
+    headers: RSRequestHeaders;
+}
+/* Prevents overriding RSRequestConfig values */
+export type AdvancedConfig = Omit<
+    AxiosRequestConfig,
+    "url" | "method" | "headers"
+>;
+/* Safe endpoint extraction */
+const extractEndpoint = (api: API, key: string): Endpoint => {
+    const endpoint: Endpoint | undefined = api.endpoints.get(key);
+    if (!endpoint)
+        throw Error(`You must provide a valid endpoint key: ${key} not found`);
+    return endpoint;
+};
+/* Handles generating the config from parameters */
+export const createAxiosConfig = <P extends RSUrlParams>(
+    api: API,
+    endpointKey: string,
+    method: Method,
+    token?: string,
+    organization?: string,
+    parameters?: P,
+    // Allows us to use more of AxiosRequestConfig if we want
+    advancedConfig?: AdvancedConfig
+): RSRequestConfig => {
```

Comment:
Todo tomorrow: forgot to add method validation in here! Should probably do a while validation run and throw if anything is missing or invalid.